chain-impl-mockchain/src/smartcontract/mod.rs

```diff
@@ -0,0 +1,47 @@
+//! EVM Smart Contract transactions
+
+use chain_core::mempack::Readable;
+#[cfg(feature = "evm")]
+use chain_evm::{Configuration, Environment, GasLimit};
+
+use crate::transaction::Payload;
+
+pub enum Contract {
+    EVM {
```

Comment:
I think the whole variant should be put behind the feature gate, if it's practical.